core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -845,6 +852,13 @@ private[spark] object Utils extends Logging {
     customHostname.getOrElse(localIpAddressHostname)
   }
 
+  /**
+   * Get the local machine's URI.
+   */
+  def localHostURI(): String = {
```

Comment:
Is this intended to be an IP address or host name? looks like IP address. But then some of the changes above are changing a host name to an IP address. That might be OK but it's not obvious? what's the motivation there.